### DISCUSSION TITLE
Home | Drupal Book

### DISCUSSION THREAD

**Discussion by giscus** [selwynpolit/d9book#65] [Q&A]
# Home | Drupal Book

Drupal at your fingertips

https://selwynpolit.github.io/d9book/

**Comment by carlosilipo77** [comment]
Hi,
This is a great resource/book.  Always use the KISS principle, ( keep it simple stupid). I just bought the book on module development and I can’t make out heads or tails.  I am one of those people that needs to know everything( how does that work).

Just remember, if you are pitching your book to beginners, then it needs lots of explanations, or if seasoned drupal users, then it can be more cryptic. 

Keep up the good work. 
Carlo

**Reply by selwynpolit** [reply]
Thanks Carlo.  I'm glad you like it.  I really struggled initially with who my audience was and eventually settled on experienced Drupal developers rather than folks new to Drupal development.  I've seen many books that start with the basics and I don't think I want to do that.  

You'll see some chapters e.g. the Forms API where I created a whole section called [The basics of implementing forms](https://selwynpolit.github.io/d9book/forms#the-basics-of-implementing-forms) where I explained at a more beginner level.  I think it would be useful to add a section like that to each chapter but I'm not sure I'm the guy for the job.  ;-)

Have you had a chance to take a peek at the [Learning chapter](https://selwynpolit.github.io/d9book/learn)?  It does reference lots of other books/videos/podcasts etc that might be helpful to you.

**Comment by carlosilipo77** [comment]
Hi again,
I really like how you add more resources to the work. For example you give a like to the drupal.org site with the corresponding topic. Well done again. Hopefully this template will be adopted by more drupal writers.

**Comment by carlosilipo77** [comment]
Just a suggestion about yr book format. I really like Rod from ostraining.com how they setup the page format on various topics, especially the screenshots. What I like the format is logical - just how I would bake a cake.

The other one is of note is webwash.net. The format is worthy to checkout.

**Reply by selwynpolit** [reply]
I agree that the webwash design is very clean.

**Comment by carlosilipo77** [comment]
There seems to heaps of resources on SSO’s online. 

https://www.drupal.org/docs/contributed-modules/saml-sp-single-sign-on-sso-saml-service-provider-by-miniorange/guide-for-drupal-single-sign-on-sso-using-google-apps-as-identity-provider-idp

And,

https://www.drupal.org/docs/contributed-modules/saml-sp-single-sign-on-sso-saml-service-provider/guide-for-drupal-single-sign-on-sso-using-simplesaml-as-identity-provider-idp

**Reply by carlosilipo77** [reply]
https://youtu.be/7RrFhC-e8QA

Saml

**Comment by lostcarpark** [comment]
This is a fantastic resource. My only problem is that I keep far too many browser tabs open. It would be great if this had a more distinctive favicon to make it easier to find. What would be the best way to contribute a new icon?

**Reply by selwynpolit** [reply]
Thanks.  Glad you like it.  It looks like the icon is in the assets/images folder.  It is called favicon.ico.  I think you should be able to fork the branch and replace it.  Then create a pull request.  You can always email me a fantastic new icon and I'll add it. ;-))

**Comment by waptug** [comment]
Great work. Loved it. I was inspired to create resource tool for this to publish this in a PDF format that is dynamically generated using Python 3.  https://github.com/waptug/Drupal9BookPDFPublisher.git Have a look. Hope you find it useful.

**Comment by danylevskyi** [comment]
Thank you for the great work!
